internal/discover/symlinks.go

```diff
@@ -110,3 +130,50 @@ func (d additionalSymlinks) isDriverLibrary(libraryName string, filename string)
 	match, _ := filepath.Match(pattern, filename)
 	return match
 }
+
+func (d *additionalSymlinks) getDotSoSymlinks(libraryPath string) ([]string, error) {
+	libraryName := filepath.Base(libraryPath)
+	if !d.isDriverLibrary("*", libraryName) {
+		return nil, nil
+	}
+
+	var soSymlinks []string
+	lib, err := elf.Open(libraryPath)
+	if err != nil {
+		return nil, err
+	}
+	defer lib.Close()
+
+	sonames, err := lib.DynString(elf.DT_SONAME)
+	if err != nil {
+		return nil, err
+	}
+
+	var sonameLinkPath string
+	for _, soname := range sonames {
+		if soname == libraryName {
+			continue
+		}
+		linkPath := filepath.Join(filepath.Dir(libraryPath), soname)
+		if sonameLinkPath == "" {
+			sonameLinkPath = linkPath
+		}
+		s := Symlink{
+			target: libraryName,
+			link:   linkPath,
+		}
+		soSymlinks = append(soSymlinks, s.String())
+	}
+
+	if sonameLinkPath != "" {
+		sonameLinkPathExt := filepath.Ext(sonameLinkPath)
+		soLinkPath := strings.TrimSuffix(sonameLinkPath, sonameLinkPathExt)
+		s := Symlink{
+			target: filepath.Base(sonameLinkPath),
+			link:   soLinkPath,
+		}
+		soSymlinks = append(soSymlinks, s.String())
+	}
```

Comment:
So you walk through all sonames, and add a symlik for them.

And at the same time you see if there was at least one soname that matched, and if there was, you store it in `sonameLinkPath` so that you can strip off the final version number from it and just additionally include the base `.so` symlink.

Is that what is going on here?

Reply:
Yes, for a given library, we extract the sonames and create symlinks for them (if they don't match the original library name). For the first soname we additionally strip off the numbered suffix and create a `.so` symlink too. 

One thing that we may want to do is explicitly match on `*.so` instead of making assumptions about the soname that we're linking to.

Reply:
I don't understand this part:
>  For the first soname we additionally strip off the numbered suffix and create a .so symlink too

Is it actually possible to have multiple SONAMEs set?

Reply:
The [`elf.h` manpage](https://man7.org/linux/man-pages/man5/elf.5.html) states:

>               DT_SONAME
                     String table offset to name of shared object

Which refers to a singular. The issue is that the `DynString` function returns a slice to handle other case. I would therefore assume that it's unexpected if we have MULTIPLE sonames. Let me update the implementation to check the length specifically.